### DISCUSSION TITLE
Address Input

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#1387] [Feedback]
Use this place to discuss the [address input pattern](https://ons-design-system.netlify.com/patterns/addresses) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
The address input needs to be made more configurable to allow only some fields being used. The current conditions used in the macro aren't compatible with Jinja so these need to be checked also.

**Comment by bameyrick** [comment]
We haven't considered validation for the address input with its multiple fields. We need to look into how we will handle mandatory field errors on this input.

**Comment by jrbarnes9** [comment]
Typo in `legendClasses` on 'Asking for an organisation or company’s address' example.

**Comment by jrbarnes9** [comment]
Add documentation about the use of the footer variant with 'OS address data terms and conditions' when using this pattern.

**Comment by jrbarnes9** [comment]
Address lookup input error panel missing border style
![image (1)](https://user-images.githubusercontent.com/43346934/112851305-c50d0a80-90a2-11eb-9a72-ac831ce8cc7e.png)

**Reply by jrbarnes9** [reply]
#1456 resolved

**Comment by jrbarnes9** [comment]
Focus state is lost when traversing between autosuggest results panel and 'I can't find my address' link

![autosuggest-focus](https://user-images.githubusercontent.com/43346934/182650731-0f07a3d5-233f-4fe5-870a-436510d9cbaf.gif)

**Reply by jrbarnes9** [reply]
Raised #2367

**Comment by AlanHONS** [comment]
Non-editable addresses example shown in Service Manual site is broken and doesn't function. 

Either inline within Service Manual https://service-manual.ons.gov.uk/design-system/patterns/addresses or when viewing full page example https://service-manual.ons.gov.uk/design-system/patterns/addresses/example-address-input-non-editable

![image](https://github.com/ONSdigital/design-system/assets/114412177/e27a3646-dd7e-4fb7-9b1c-c4267287ed3a)

**Reply by daniellecorke** [reply]
Hi Alan, thanks for letting us know about this. I've raised #3032 to look into it.

**Reply by daniellecorke** [reply]
Very happy to report that after much hard work #3032 is resolved. The team have done some amazing work to create a working example of the address input which was included in today's release. An example of how the address input is displayed along with guidance regarding the ONS Address Index API can be found here: https://service-manual.ons.gov.uk/design-system/components/address-input.

**Reply by AlanHONS** [reply]
Thank you for this update @daniellecorke 
Checking out https://service-manual.ons.gov.uk/design-system/components/address-input I do have some comments for you to consider though:

1. It appears that shown matches are random, not consistent and not related to input characters?

As if I enter 'abc' I initially got 4x matches of:
![image](https://github.com/ONSdigital/design-system/assets/114412177/bdac9b88-6875-4b2c-a1f6-c666322d7095)

But if I deleted end 'c' and then retyped it back to 'abc' I then got 0 matches:
![image](https://github.com/ONSdigital/design-system/assets/114412177/8888c71e-5c2d-4715-8e4a-13a12831479f)

If I then refresh the page and enter 'abc' I get different 3x matches of:
![image](https://github.com/ONSdigital/design-system/assets/114412177/b277b1f9-39ce-4a83-9657-8838533aed4c)

2. Is it expected that there's no relation between entered characters and matching results? 
Sure while an example API it should only return results which contain entered letters? (See above 4x and 3x results screenshots)

3. FYI On initial load of that page I did get an Error state message below Non-editable address finder example entry field. 
But that went away after page refresh and I haven't seen that again atm

**Reply by alessioventuriniAND** [reply]
Hello Alan, 

The mock api that has been built only returns complete random results of fake addresses. The behaviour you have witnessed is totally normal and expected. 

The purpose of that api is to show a working address input which allows you to see how the autosuggest appears upon searches and how the address input changes once an address is selected. The content displayed in the input is of no relevance in this case. 

Regarding the error state on first load, that is something I will need to look into and troubleshoot. 

Happy to help if you have any other questions.

**Comment by Paul-Joel** [comment]
Not sure if this is the correct place any more, but there appears to be an issue in the non-editable version.
When AIMS recognises a postcode, it adds an entry for the postcode, with a link to show all the entries for the postcode as a link in the text result, however the link is not clickable, as clicking it selects the item rather than clicking the link.
In the image, the (29 addresses) should be clickable, and not part of the click area for the option
This needs to be resolved for Census
<img width="1066" height="730" alt="image" src="https://github.com/user-attachments/assets/7a43943c-7193-4b7b-9b0b-36233a901022" />

**Comment by Olatunde-Alalade-ONS** [comment]
Hi Paul. yes it has moved. But we will be updating the Service Manual soon with the new template. In the meantime, I have created this ticket [ONSDESYS-981](https://officefornationalstatistics.atlassian.net/browse/ONSDESYS-981) Please have a look and confirm that I have captured the details correctly.  Please feel free to amend if required.

**Reply by Paul-Joel** [reply]
Ticket looks good, thanks

**Reply by Olatunde-Alalade-ONS** [reply]
closing this one. We'll monitor the one.
